packages/apps/plugins/plugin-function/src/components/TriggerEditor.tsx

```diff
@@ -29,15 +39,40 @@ export const TriggerEditor = ({ space, trigger }: { space: Space; trigger: Funct
     [trigger.function, functions],
   );
 
-  // Initialize meta.
+  useEffect(() => {
+    if (space) {
+      void space.db.schemaRegistry
+        .getAll()
+        .then((schemas) => {
+          // TODO(Zan): We should solve double adding of stored schemas in the schema registry.
+          state.schemas = distinctBy([...state.schemas, ...schemas], (s) => s.typename).sort((a, b) =>
+            a.typename < b.typename ? -1 : 1,
+          );
+        })
+        .catch();
+    }
+  }, []);
+
+  // Keen an enriched version of the schema in memory so we can share it with prompt editor.
+  useEffect(() => {
+    const spec = trigger.spec;
+    if (spec.type === 'subscription') {
+      if (spec.filter && spec.filter.length > 0) {
+        const type = spec.filter[0].type;
+        const foundSchema = state.schemas.find((schema) => schema.typename === type);
+        if (foundSchema) {
+          state.selectedSchema[trigger.id] = foundSchema;
+        }
+      }
+    }
+  }, [JSON.stringify(trigger.spec), state.schemas]);
```

Comment:
Would love a signal utility that deeply watches a collection.

Reply:
yes -- we must do this -- there's a couple of hacks around the place to do this kind of thing.